dev/AppLifecycle/AppInstance.cpp

```diff
@@ -33,30 +34,37 @@ namespace winrt::Microsoft::Windows::AppLifecycle::implementation
         // Push past the '----' commandline argument prefix.
         argsStart += 4;
 
-        // We explicitly use find_first_of here, so that the resulting data may contain : as a valid character.
-        auto argsEnd = commandLine.find_first_of(L":", argsStart);
+        auto argsEnd = commandLine.find_first_of(L" ", argsStart);
+
+        // Separate the argument from any behind it on the command-line.
+        std::wstring argument;
         if (argsEnd == std::wstring::npos)
         {
-            return { L"", L"" };
+            argument = commandLine.substr(argsStart);
         }
-
-        if (argsStart > argsEnd)
+        else
         {
-            throw std::overflow_error("commandLine");
+            if (argsStart > argsEnd)
+            {
+                throw std::invalid_argument("commandLine");
+            }
+
+            argument = commandLine.substr(argsStart, (argsEnd - argsStart));
         }
 
-        auto argsLength = argsEnd - argsStart;
-        auto dataStart = argsEnd + 1;
+        // We explicitly use find_first_of here, so that the resulting data may contain : as a valid character.
+        auto argsDelim = argument.find_first_of(L":");
+        if (argsDelim == std::wstring::npos)
+        {
+            return { argument, L"" };
+        }
 
-        return { commandLine.substr(argsStart, argsLength), commandLine.substr(dataStart) };
+        return { argument.substr(0, argsDelim), argument.substr(argsDelim + 1) };
     }
```

Comment:
Creating multiple temporary strings seems like unnecessary effort. Could just work in indices and then do the substing at the end (or perhaps never and just work with wstring_views).

```cpp
std::tuple<std::wstring_view, std::wstring_view> ParseCommandLine(PCWSTR rawCommandLine)
{
    auto commandLine = std::wstring_view(rawCommandLine);

    auto argsStart = commandLine.rfind(c_argumentPrefix);
    if (argsStart != std::wstring::npos)
    {
        // push past the "----" argument prefix
        commandLine = commandLine.substr(argsStart + 4);
    }
    else
    {
        commandLine = {};
    }

    // Stop at first space, if any.
    auto argsEnd = commandLine.find_first_of(L' ');
    if (argsEnd != std::wstring::npos)
    {
        commandLine = commandLine.substr(0, argsEnd);
    }

    auto delimiter = commandLine.find_first_of(L':');
    if (delimiter == std::wstring::npos)
    {
        // No delimeter means it all belongs to the first half.
        return { commandLine, {} };
    }

    return { commandLine.substr(0, delimiter), commandLine.substr(delimiter + 1) };
}

```

Now you don't have to worry about `find_first_of` returning an end point earlier than the start point, because that is simply inexpressible in a `wstring_view`. No memory allocation required; we just operate in string views.

Reply:
Is there a reason why we chose not to switch to the more efficient (and in my mind simpler) `wstring_view`?

Reply:
> Is there a reason why we chose not to switch to the more efficient (and in my mind simpler) `wstring_view`?

Doh! A new issue was created as a follow up for several things that were preexisting AppLifecycle code. This one?
[Address AppLifecycle comments from Push Notifications PR #926](https://github.com/microsoft/ProjectReunion/issues/926)

I tried to comment on just changes in this PR. Comments on anything preexisting can be handled via a separate PR if you like

Reply:
Okay, that's what happened. A comment to that effect would have been appreciated. "Added to issue <blah> and resolving here." As it stood, it looked like the issue was simply resolved without being addressed.